### PR TITLE
Update kustomize build github workflow

### PR DESCRIPTION
**Pull request by natifridman**
- Update Kustomize version
- Update checkout action to v4
- Use cnf-features-deploy `release-4.13` branch instead of main

### CODE REVIEW

File: .github/workflows/kustomize-lint.yaml [hunk 1]
```diff
@@ -32,7 +33,7 @@ jobs:
           cd /home/runner/work/vse-carslab-hub/vse-carslab-hub/cnf-features-deploy/ztp/policygenerator-kustomize-plugin
           make build KUSTOMIZE_DIR=/home/runner/kustomize POLICYGEN_KUSTOMIZE_DIR=/home/runner/kustomize/plugin/ran.openshift.io/v1/policygentemplate
       - name: Code Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
```
Comment (novacain1) [inline .github/workflows/kustomize-lint.yaml:36]: Should we be concerned that this line is v3 in the 2nd main (cars2) branch?

Comment (natifridman) [reply .github/workflows/kustomize-lint.yaml:36]: Not urgent, but we should update it there as well

Comment (novacain1) [reply .github/workflows/kustomize-lint.yaml:36]: will tackle next time I do a PR there.

**Review by novacain1 — APPROVED** [review]

**Pull request merged**